### ISSUE TITLE
Implement Portal Farms in Vue3

### ISSUE THREAD

**Issue by zaelgohary** [threefoldtech/tfgrid-sdk-ts#1017]

**Comment by zaelgohary** [comment]
On hold till the design is completed.

**Comment by zaelgohary** [comment]
Blocked on https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1090

**Comment by zaelgohary** [comment]
Added the missing methods from tfgridmodule in chain & client.

**Comment by zaelgohary** [comment]
Added getUserNodes & created a store for grid client in order to avoid creating many instances in Playground.

**Comment by zaelgohary** [comment]
Still working on the details in both data tables.

**Comment by zaelgohary** [comment]
Removed getUserNodes and worked on farm details table.

**Comment by zaelgohary** [comment]
Worked on add farm IP and IP validation.

**Comment by zaelgohary** [comment]
Fixed Add/edit stellar. Added node details.

**Comment by zaelgohary** [comment]
Added nodes tables actions.

**Comment by zaelgohary** [comment]
Fixed uptime & working on implementing another date-picker in node stats section as the current one is incompatible with the Vuetify version we're using.

**Comment by zaelgohary** [comment]
Updated Vuetify version and fixed issues in farms & nodes tables. Still working on node stats.

**Comment by zaelgohary** [comment]
Reverted Vuetify update changes as they were unnecessary. Finished minting details. Ths issue is almost finished, minus some styling details and validation bugs.

**Comment by zaelgohary** [comment]
Worked on some styling and validation issues.

**Comment by zaelgohary** [comment]
PR is ready for review.

**Comment by zaelgohary** [comment]
Edit pubic config validation, Added loading to all buttons. Update tables once data is changed. Edit addConfig in chain & grid.

**Comment by zaelgohary** [comment]
Still working on some styling issues and PR comments.

**Comment by zaelgohary** [comment]
Changed farm table to be server-side, fixed search farm by to filter by ID & name. disable sorting in both tables. Changed farm details view to be in card details as other pages.

**Comment by zaelgohary** [comment]
I'm still working on resolving the comments. On hold till Grid Proxy is updated.

**Comment by zaelgohary** [comment]
Still working on comments.

**Comment by zaelgohary** [comment]
Edited IP validations and fixed props type, updated stellar once edited.

**Comment by zaelgohary** [comment]
Added a debounce function to update the table once stellar is edited. Toast colors should be changed in a different issue as it's been discussed with @ehab-hassan . PR is ready for review.

**Comment by zaelgohary** [comment]
Added tooltips to add_ip dialogue inputs. Added a check before setting the uptime.

**Comment by zaelgohary** [comment]
Worked on validation, styling, type checking & table updating issues. WIP: some bugs in public config pop up.

**Comment by zaelgohary** [comment]
Still working on some issues in the public config. 

Fixed some issues in add IP & add range & table update. Batched range IPs instead of creating extrinsic one by one, fixed gateway validation, fixed delete IP dialogue, refactored node.ts to use already made types from mintings, edited farms & nodes types to use from grid proxy. Edited grid client validation for set extra fee.

**Comment by zaelgohary** [comment]
WIP: 
Fixed some issues in public config pop-up but still working on input validation issues as some validations are being done even though inputs are empty. 

Remaining Issues:
- Public config optional inputs validation
- Download minting receipts in farm details

**Comment by zaelgohary** [comment]
Work completed:

- fixed public config validation
- added download farm minitng receipts

**Comment by A-Harby** [comment]
Verified,
Devnet, 28d6c07.

All the portal pages are implemented in the new Dashboard UI and are working fine; all the other bugs have been reported and are being fixed.
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/9215c1b0-ff62-461f-8ec5-646edd737ed8)
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/38952157-174c-43e6-a388-0c8f15a816dc)
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/d37cc9b6-6dc9-4b33-85d8-97b62409cfbc)
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/4bcb2dd2-05f9-4b22-83fa-dae59f91c1fa)
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/b8532342-f139-41b0-a739-5c0e4e5c6c06)
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/ed16288a-1737-49b9-8e14-36825d120f26)
